setup.py

```diff
@@ -50,7 +51,22 @@
 if os.environ.get("TEST_JUJU3"):
     install_require.append('juju')
 else:
-    install_require.append('juju<3.0.0')
+    try:
+        version = subprocess.check_output(['juju', '--version'],
+                                          universal_newlines=True).strip()
+        print('juju --version ->', version)
+        if int(version[0]) >= 3:
+            (major, minor) = version.split('.')[0:2]
```

Comment:
This will fail whenever minor version cropped into double digit like 4.10

```
(major, minor, *_) = version.split('.')
juju_ver_n = "%s.%s" % (major, minor)
juju_ver_n1 = "%s.%s" % (major, int(minor) + 1)
```

Reply:
I don't think that will make that piece of code fail:

```
>>> version = "4.10.6-genericlinux-amd64"
>>> (major, minor) = version.split('.')[0:2]
>>> major
'4'
>>> minor
'10'
```

Reply:
AAh such a silly mistake from me, you are right

Reply:
missing `''` for `.` in the new change